src/detect-engine.c

```diff
@@ -691,6 +691,9 @@ static void AppendAppInspectEngine(DetectEngineCtx *de_ctx,
     new_engine->sm_list = t->sm_list;
     new_engine->sm_list_base = t->sm_list_base;
     new_engine->smd = smd;
+    if (t->v2.Callback == DetectEngineInspectBufferGeneric) {
```

Comment:
this means only a subset of sticky buffers can match on this, right? E.g. dns.query or file.data won't be supported, but the rules won't be rejected either?

Reply:
For me dns.query is a multi buffer, not a sticky buffer...
What rule should be rejected ?
What would you expect the behavior to be ?
Currently, negated content on a multi buffer matches if one of the buffer does not have the content, not if all of them do not have the content... right ?
Is there some doc to update ?

I thought `file.data` was a streaming buffer, and not a sticky buffer...
What rule should be rejected ?

Reply:
stickybuffer is just  meant to reflect all the keywords that precede content and friends in the rule syntax, so it includes dns.query, file.data, etc

Reply:
So, I guess the good way to do this for multi-buffer is a rewrite to have `DetectEngineInspectMultiBufferGeneric` which will handle `match_on_null`
And so, to have `InspectionBufferGetDataPtr` augmented by an argument which is the local id of the buffer to get

What do you think ?

Reply:
cf https://redmine.openinfosecfoundation.org/issues/6575